client/reader/onboarding/subscribe-modal/index.tsx

```diff
@@ -1,17 +1,88 @@
+import { LoadingPlaceholder } from '@automattic/components';
+import { useQuery } from '@tanstack/react-query';
 import { Modal } from '@wordpress/components';
 import React from 'react';
+import { useSelector } from 'react-redux';
+import ConnectedReaderSubscriptionListItem from 'calypso/blocks/reader-subscription-list-item/connected';
+import wpcom from 'calypso/lib/wp';
+import { getReaderFollowedTags } from 'calypso/state/reader/tags/selectors';
 
 interface SubscribeModalProps {
 	isOpen: boolean;
 	onClose: () => void;
 }
 
+interface CardData {
+	ID: number;
+	site_ID: number;
+	feed_ID: number;
+	site_URL: string;
+	site_name: string;
+}
+
+interface Card {
+	type: string;
+	data: CardData;
+}
+
 const SubscribeModal: React.FC< SubscribeModalProps > = ( { isOpen, onClose } ) => {
+	const followedTags = useSelector( getReaderFollowedTags ) || [];
+	const followedTagSlugs = followedTags.map( ( tag ) => tag.slug );
+	followedTagSlugs.sort();
+
+	const { data: recommendedSites = [], isLoading } = useQuery( {
+		queryKey: [ 'reader-onboarding-recommended-sites', followedTagSlugs ],
+		queryFn: () =>
+			wpcom.req.get(
+				{
+					path: `/read/tags/cards`,
+					apiNamespace: 'wpcom/v2',
+				},
+				{
+					tags: followedTagSlugs,
+					site_recs_per_card: 6, // This does not seem to have any effect. 8 results are always returned.
```

Comment:
I copied this from @Addison-Stavlo [prototype](https://github.com/Automattic/wp-calypso/pull/84935/files#diff-cff253bd8ba10e74018ef69d5288ea8517a47186f54414732b16dd4e77cd4663R24), but it doesn't seem to change the number of results returned. Maybe I'm misunderstanding what `site_recs_per_card` does?

Reply:
> it doesn't seem to change the number of results returned. Maybe I'm misunderstanding what site_recs_per_card does?

I don't think its supposed to change the number of overall results (cards) returned, but how many recommendations are added to the `recommended_blogs` card type itself. Different recommendation cards are supposed to be filtered in between posts returned in the list, some for tags, blogs, and feeds.

The cards returned will have types (as you have seen below), standard `post` types, `interests_you_may_like` types (recommended tags), `recommended_blog` types, etc. And the param is supposed to change how many recommendations are on the `recommended_blog` type card (similar to `tag_recs_per_card` changes the number of tags listed on the card of type `interests_you_may_like`).

Im not currently seeing any of the `recommended_blog` type cards here, just the one tag card and a bunch of standard posts.
<img width="678" alt="Screenshot 2024-10-11 at 10 04 38 AM" src="https://github.com/user-attachments/assets/1604e59a-ba9b-4c1d-a528-f692653c678a">

We may need to investigate the endpoint to figure out why the recommended_blogs card isn't being inserted?
`/wp-content/rest-api-plugins/endpoints/read-tags-cards.php`

Reply:
After investigating a little more, it looks like D159312 may be the culprit that has prevented the recommended blogs cards from appearing in the response.

It adds a line to the function getting the recommended blogs:
`$global_tag_recommendations_sites->enable_filter_new_or_empty_blogs();` that seems to be filtering everything out in my testing. If I comment this line out, I start to see recommended_blogs cards again:

<img width="416" alt="Screenshot 2024-10-11 at 10 19 04 AM" src="https://github.com/user-attachments/assets/1d30cff5-00c4-40c1-af07-1ed5247318b7">